src/typescript-worker/completions.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import Fuse from 'fuse.js';
+import type {
+  GetCompletionsAtPositionOptions,
+  SymbolDisplayPart,
+} from 'typescript';
+import {
+  EditorCompletion,
+  EditorCompletionDetails,
+  EditorCompletionMatch,
+  EditorToken,
+  WorkerConfig,
+} from '../shared/worker-api.js';
+import {getWorkerContext} from './worker-context.js';
+
+/**
+ * Query completions from the Language Service, and sort them by
+ * relevance for user to use.
+ * */
+export const queryCompletions = async (
+  filename: string,
+  fileContent: string,
+  tokenUnderCursor: EditorToken,
+  cursorIndex: number,
+  config: WorkerConfig
+): Promise<EditorCompletion[]> => {
+  const workerContext = getWorkerContext(config);
+
+  const languageService = workerContext.languageServiceContext.service;
+  const languageServiceHost = workerContext.languageServiceContext.serviceHost;
+  const searchWordIsPeriod = tokenUnderCursor.string === '.';
+
+  const options = {} as GetCompletionsAtPositionOptions;
+  if (searchWordIsPeriod) {
+    options.triggerCharacter = '.';
+  }
+
+  // Update language service status so that the file is up to date
+  languageServiceHost.updateFile(filename, fileContent);
```

Comment:
There always is the _potential_ that some kind of a bug could arise from this, but As it is now, (especially with fixing the call to use the `updateFileContentIfNeeded` -function), I don't see it causing too much trouble in the present. We should however keep it in mind, that it might cause something down the line.

I'll add a TODO to this code line, to remind ourselves about it